src/slider/slider.js

```diff
@@ -0,0 +1,225 @@
+    const {thumbRefs} = this.state;
+    const [Root, rootProps] = getOverrides(overrides.Root, StyledRoot);
+    const [Axis, axisProps] = getOverrides(overrides.Axis, StyledAxis);
+    const [AxisRange, axisRangeProps] = getOverrides(
+      overrides.AxisRange,
+      StyledAxisRange,
+    );
+    const [Thumb, thumbProps] = getOverrides(overrides.Thumb, StyledThumb);
+    const [Tick, tickProps] = getOverrides(overrides.Tick, StyledTick);
+    const [TickBar, tickBarProps] = getOverrides(
+      overrides.TickBar,
+      StyledTickBar,
+    );
+    const sharedProps = this.getSharedProps();
+    return (
+      <Root {...sharedProps} onClick={e => this.onAxisClick(e)} {...rootProps}>
+        <Axis {...sharedProps} {...axisProps}>
+          {value.map((thumb, $index) => (
+            <React.Fragment key={$index}>
+              <AxisRange $index={$index} {...sharedProps} {...axisRangeProps} />
+              <Thumb
+                $ref={thumbRefs[$index]}
+                key={$index}
+                $index={$index}
+                {...thumbProps}
+                {...sharedProps}
+                onMouseDown={e => this.onMouseDown(e, $index)}
+              />
+            </React.Fragment>
+          ))}
+        </Axis>
+        <TickBar {...sharedProps} {...tickBarProps}>
+          {range.map((tick, $index) => (
+            <Tick key={$index} $index={$index} {...sharedProps} {...tickProps}>
+              {tick}
+            </Tick>
+          ))}
+        </TickBar>
+      </Root>
+    );
+  }
+
+  onMove(movementX: number, e: SyntheticEvent<HTMLElement> | MouseEvent) {
+    const {currentThumb, currentMove} = this.state;
+    const move = currentMove + movementX;
+    const {$value, $max, $min, $isRange} = this.getSharedProps();
+    const value = $value.slice();
+    let {onChange, step} = this.props;
+    const axisSizeInPixels = parseInt(
+      window.getComputedStyle(this.domNode).width,
+    );
+    const axisSize = $max - $min;
+    step = step ? Slider.scale(step, axisSize, axisSizeInPixels) : step;
+    const scaledStep = step ? (move >= 0 ? step : -step) : move;
+    if (Math.abs(move) - Math.abs(scaledStep) >= 0) {
+      const scaledMove = Slider.scale(scaledStep, axisSizeInPixels, axisSize);
+      const max = !currentThumb && $isRange ? value[1] : $max;
+      const min = currentThumb && $isRange ? value[0] : $min;
+      value[currentThumb] = Slider.clampIt(
+        value[currentThumb] + scaledMove,
+        max,
+        min,
+      );
+      this.setState({currentMove: 0});
+    } else {
+      this.setState({currentMove: move});
+    }
+    onChange(e, {type: STATE_CHANGE_TYPE.value, value});
+  }
+
+  onAxisClick(e: MouseEvent) {
+    const {value} = this.props;
+    const {thumbRefs} = this.state;
+    // only for one thumb feature
+    //$FlowFixMe
+    if (value.length === 1 && e.currentTarget.parentNode) {
+      const thumb = thumbRefs[0].current;
+      const thumbWidth = 32;
```

Comment:
where does this constant comes from? should we move it out to a contants file?

Reply:
moved to constants and shown link to Thumb itself. I don't like magic numbers myself.